### PR TITLE
MB-3243 Integrate Orders document viewer

### PR DESCRIPTION
**Pull request by Ryan-Koch**
## Description

This PR adds the ability for a TOO to view orders documents that the user uploaded.

## Reviewer Notes

Steps to test:
1. Run the setup commands in the section below
2. Visit http://officelocal:3000/devlocal-auth/login
2. Click the `Login` button next to `too_role@office.mil (PPM office)`
3. Click on the first `MoveOrderID` in the list
4. Change the url from http://officelocal:3000/moves/[move_id]/details to http://officelocal:3000/moves/[move_id]/orders

Expected Result: you should see a test.pdf file

By default, the seed data only uploads PDFs. To test other image types, follow these steps:

1. Sign in to http://milmovelocal:3000 as a service member
2. If you haven't created a move yet, go through onboarding and create one. When it's time to upload your orders documents, choose a PNG, JPG, or GIF. Please make sure they don't contain any PII. If you've already created a move, you can edit it and upload additional orders documents.
3. Visit http://officelocal:3000/devlocal-auth/login
4. Click the `Login` button next to `too_role@office.mil (PPM office)`
5. Click on the Move that you created as a service member.
6. Change the url from http://officelocal:3000/moves/[move_id]/details to http://officelocal:3000/moves/[move_id]/orders

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
make db_dev_e2e_populate
make server_run
make client_run (in a separate terminal tab)
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3243) for this change

## Screenshots

Screen Shot 2020-09-10 at 2 59 48 PM

### CODE REVIEW

File: swagger/ghc.yaml [hunk 1]
```diff
@@ -1185,6 +1185,58 @@ paths:
...
         - ppm_office_users
       summary: Updates status of a payment request by id
+  /documents/{documentId}:
+    get:
+      summary: Returns a document
+      x-swagger-roles:
+        - transportation_invoicing_officer
+        - transportation_ordering_officer
+        - contracting_officer
+        - ppm_office_users
```
Comment (monfresh) [inline swagger/ghc.yaml:1195]: We're not using these `x-swagger-roles` anymore, so we can remove them. See PR #4688.

File: src/hooks/queries.js [hunk 2]
```diff
@@ -95,3 +103,26 @@ export const useMoveTaskOrderQueries = (moveOrderId) => {
     isSuccess,
   };
 };
+
+export const useOrdersDocumentQueries = (moveOrderId) => {
```
Comment (duncan-truss) [inline src/hooks/queries.js:107]: Just putting it out there that the internal api behaves differently including documents in the orders.showOrders query unlike the ghc one.  We'll have some work to do one way or the other.

File: src/components/DocumentViewer/DocumentViewer.jsx [hunk 3]
```diff
@@ -24,6 +24,9 @@ const DocumentViewer = ({ files }) => {
   const [menuIsOpen, setMenuOpen] = useState(false);
 
   const selectedFile = files[parseInt(selectedFileIndex, 10)];
+  if (selectedFile.contentType === 'application/pdf') {
```
Comment (duncan-truss) [inline src/components/DocumentViewer/DocumentViewer.jsx:27]: The image support PR was just merged in if we could add conditionals here for jp(e)g and png it should close that loop.

File: src/hooks/queries.js (hunk 2 above)
Comment (monfresh) [reply src/hooks/queries.js:107]: Do you know why we have 2 separate APIs for this? It looks like a lot of the code that was added to the GHC API in this PR is identical to the one in the internal API.

**Review by brooksr4 — COMMENTED** [review]
I get an error for each Move that I attempted to open the orders.
Here are the steps that I took:
Step 1:
```
make db_dev_e2e_populate
make server_run
make client_run
```
Step 2: Sign in as TOO in the office interface
Step 3: Open a move
Step 4: change the url from `http://officelocal:3000/moves/[move_id]/details` to `http://officelocal:3000/moves/[move_id]/orders`

and I got the following error for each move available:
image

**Comment by monfresh** [comment]
Thanks for the bug report. I will take a look.

File: swagger/ghc.yaml [hunk 4]
```diff
@@ -2181,6 +2232,25 @@ definitions:
       eTag:
         type: string
     type: object
+  DocumentPayload:
```
Comment (duncan-truss) [inline swagger/ghc.yaml:2235]: Does this payload differ from just being the Document model?  Payload to me means something was added or subtracted from the model.

File: src/pages/Office/MoveOrders/MoveOrders.jsx [hunk 5]
```diff
@@ -1,45 +1,42 @@
...
-    },
-  ];
+const MoveOrders = ({ match }) => {
+  const { moveOrderId } = match.params;
+  const {
+    documents,
+    // eslint-disable-next-line no-unused-vars
+    isLoading,
+    // eslint-disable-next-line no-unused-vars
+    isError,
```
Comment (duncan-truss) [inline src/pages/Office/MoveOrders/MoveOrders.jsx:17]: Usually we return early if isLoading or isError have a value before the render method. Here is an example so we don't need to disable the lint rules https://github.com/transcom/mymove/blob/b075756c7bb85c5f105946a6e55837ef67da1af7/src/pages/Office/MoveTaskOrder/MoveTaskOrder.jsx#L102-L103

File: swagger/ghc.yaml (hunk 4 above)
Comment (monfresh) [reply swagger/ghc.yaml:2235]: It looks like Ryan added this to follow the same pattern as in the internal API. I see that the Document model has a field called `DeletedAt` which isn't part of the payload here. Also, it refers to uploads as `UserUploads` as opposed to plain `uploads` here.

**Review by duncan-truss — APPROVED** [review]
Thanks for finishing this Moncef!

I am still seeing the exception error in some of the orders: http://officelocal:3000/moves/e4b8ab70-a35c-482c-a844-ebd9e0601fc3/orders

Do you anticipate differences in s3 file handling, like the antivirus scan?

The undefined key coming back in the payload is a little weird I wonder if that is meant to be the document id?

Could worth be adding a todo to save only the filename and not the local path in the database.

**Comment by monfresh** [comment]
I wasn't able to fix the uploads for all orders in the `e2ebasic` file. It wasn't easily clear where the `UserUploader` assertion is missing. I'll open a separate story for defaulting to creating real uploads for all generated orders.

@duncan-truss Can you show me an example of the undefined key in the payload?

**Comment by duncan-truss** [comment]
> I wasn't able to fix the uploads for all orders in the `e2ebasic` file. It wasn't easily clear where the `UserUploader` assertion is missing. I'll open a separate story for defaulting to creating real uploads for all generated orders.
> 
> @duncan-truss Can you show me an example of the undefined key in the payload?

@monfresh The undefined key you can find by clicking on the react-query devtools flower/pinwheel icon in the lower left on the orders document viewer page.  You should see the get document query and a data explorer on the right where the response json is normalized to the model name and then the id of the entity.

Screen Shot 2020-09-11 at 10 18 12 AM

**Comment by monfresh** [comment]
Thanks. I see it now. `documents` is an array, so I think the `undefined` key is superfluous. I'm not sure where it's coming from. I'll open a story to look into it.

**Review by suzubara — APPROVED** [review]
added tests for the new query hook, looks good!!

File: src/pages/Office/MoveOrders/MoveOrders.jsx [hunk 6]
```diff
@@ -1,45 +1,42 @@
...
   return (
     <div className={styles.MoveOrders}>
-      <div className={styles.embed}>
-        <DocumentViewer files={testFiles} />
-      </div>
+      {documentsForViewer && (
+        <div className={styles.embed}>
+          <DocumentViewer files={documentsForViewer} />
+        </div>
+      )}
```
[the remark this answers is not in the document]
Comment (monfresh) [reply src/pages/Office/MoveOrders/MoveOrders.jsx:28]: That's a good idea.

**Review by brooksr4 — APPROVED** [review]

**Pull request merged**